build/azDevOps/azure/air-infrastructure-data.yml

```diff
@@ -106,154 +106,101 @@ stages:
-  #       pool:
-  #         vmImage: $(pool_vm_image)
-  #       environment: prod
-  #       variables:
-  #         - name: namespace
-  #           value: "$(Environment.ShortName)-${{ variables.domain }}"
-  #       strategy:
-  #         runOnce:
-  #           deploy:
-  #             steps:
-  #               - template: ../templates/air-api-setup.yml
-  #                 parameters:
-  #                   TaskctlVersion: ${{ variables.TaskctlVersion }}
+                - template: ../templates/air-infrastructure-data-setup.yml
+                  parameters:
+                    TaskctlVersion: ${{ variables.TaskctlVersion }}
 
-  #               - task: DownloadPipelineArtifact@2
-  #                 displayName: "Download: Test Artefacts"
-  #                 inputs:
-  #                   artifact: tests
-  #                   path: $(Build.Workspace)/tests
+                - task: Bash@3
+                  displayName: "Taskctl: Infrastructure"
+                  inputs:
+                    targetType: inline
+                    script: taskctl infrastructure
+                    informationPreference: continue
+                  env:
+                    ENV_NAME: $(Environment.ShortName)
+                    # Azure Authentication
+                    CLOUD_PROVIDER: "$(cloud_provider)" # Azure
+                    ARM_CLIENT_ID: $(azure-client-id)
+                    ARM_CLIENT_SECRET: $(azure-client-secret)
+                    ARM_SUBSCRIPTION_ID: $(azure-subscription-id)
+                    ARM_TENANT_ID: $(azure-tenant-id)
+                    # Terraform Backend Configuration
+                    TF_STATE_CONTAINER: $(tf_state_container)
+                    TF_STATE_KEY: $(tf_state_key)
+                    TS_STATE_RG: $(tf_state_rg)
+                    TF_STATE_STORAGE: $(tf_state_storage)
+                    TF_BACKEND_ARGS: "key=$(TF_STATE_KEY),storage_account_name=$(TF_STATE_STORAGE),resource_group_name=$(TF_STATE_RG),container_name=$(TF_STATE_CONTAINER),subscription_id=$(azure-subscription-id),tenant_id=$(azure-tenant-id),client_id=$(azure-client-id),client_secret= $(azure-client-secret)"                    # Deployment Specific Configuration
+                    # Deployment Specific Configuration
+                    TF_VAR_core_environment: $(core_environment)
+                    TF_VAR_name_component: $(component)
+                    TF_VAR_resource_group_location: $(region)
+                    TF_VAR_create_cdn_endpoint: false
+                    TF_VAR_create_dns_record: true
+                    TF_VAR_name_company: $(company)
+                    TF_VAR_name_project: $(project)
+                    TF_VAR_name_domain: $(domain)
+                    TF_VAR_stage: $(Environment.ShortName)
+                    
+  - stage: Release
+    dependsOn:
+      - Build
+      - Prod
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'), eq(variables['create_release'], 'true'))
```

Comment:
`master` or `main`, you have both in this PR (maybe we should put this into variables). - Let's speak further

Reply:
for now have updated this to main, but lets speak to see what do we usually do